[Swift in WebKit] Work towards modularizing WebCore private headers (Part 10)

rr-codes opened the pull request:
#### aaab02499929aec8c40826df583b709382e1cf56
<pre>
[Swift in WebKit] Work towards modularizing WebCore private headers (Part 10)
<a href="https://bugs.webkit.org/show_bug.cgi?id=298068">https://bugs.webkit.org/show_bug.cgi?id=298068</a>
<a href="https://rdar.apple.com/159395829">rdar://159395829</a>

Reviewed by Aditya Keerthi.

More changes to support modularization of WebCore; details inline.

* Configurations/CommonBase.xcconfig:

Add __ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0 to GCC_PREPROCESSOR_DEFINITIONS; this is needed so that it is 0
when building with modules (it&apos;s already defined as 0 in wtf/Platform.h, but that does not affect modules).

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:

Make some headers project instead of private.

* Source/JavaScriptCore/assembler/MacroAssemblerARMv7.h:
* Source/JavaScriptCore/heap/CellAttributes.h:
* Source/WebCore/contentextensions/ContentExtensionActions.h:
* Source/WebCore/crypto/CryptoAlgorithm.h:
* Source/WebCore/page/cocoa/WebTextIndicatorLayer.h:
* Source/WebCore/platform/graphics/Color.h:
* Source/WebCore/platform/graphics/ColorSpace.h:
* Source/WebCore/platform/graphics/ColorTypes.h:
* Source/WebCore/platform/graphics/DestinationColorSpace.h:
* Source/WebCore/platform/graphics/Path.h:
* Source/WebCore/platform/graphics/transforms/AffineTransform.h:
* Source/WebCore/style/values/text-decoration/StyleTextDecorationLine.h:
* Source/WebCore/svg/SVGTRefElement.cpp:
* Source/WebCore/svg/animation/SVGSMILElement.cpp:
* Source/WebCore/testing/Internals.cpp:
* Source/WebCore/workers/service/ServiceWorkerContainer.cpp:
* Source/WebCore/workers/service/ServiceWorkerContainer.h:

- Fix some quoted imports.
- Add missing includes

* Source/WTF/Scripts/modulemap-from-tapi-filelist.py:

All modules and sub-modules should always `export *` so that transitive dependency behavior remains intact.
Consequently, update the script to add this.

* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(WriteData):
* Source/WebCore/css/MediaQueryList.cpp:
* Source/WebCore/dom/EventListenerMap.cpp:
* Source/WebCore/dom/EventTarget.cpp:
* Source/WebCore/html/HTMLAttachmentElement.cpp:
* Source/WebCore/html/ImageDocument.cpp:
* Source/WebCore/html/PDFDocument.cpp:
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
* Source/WebCore/inspector/agents/InspectorIndexedDBAgent.cpp:
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.mm:
* Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm:
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginAnnotation.mm:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginPasswordField.mm:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginTextAnnotation.mm:
* Source/WebKitLegacy/mac/DOM/DOMNode.mm:

Use the AddEventListenerOptionsInlines.h file instead of AddEventListenerOptions.h

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

* Source/WebCore/accessibility/mac/CocoaAccessibilityConstants.h:

Fix typo and missed #define.

* Source/WebCore/dom/AddEventListenerOptions.h:
(WebCore::AddEventListenerOptions::AddEventListenerOptions): Deleted.
* Source/WebCore/dom/AddEventListenerOptionsInlines.h: Copied from Source/WebCore/dom/AddEventListenerOptions.h.
(WebCore::AddEventListenerOptions::AddEventListenerOptions):

Refactor the AddEventListenerOptions slightly to avoid an #include ordering issue due to a cycle.

* Source/WebCore/dom/EventTarget.h:

Include AddEventListenerOptions instead of using a forward declaration since the full type is needed for the variant
that uses it.

* Source/WebCore/html/canvas/WebGLExtension.h:
* Source/WebCore/html/canvas/WebGLExtensionAny.h:

Add missing newline and remove unnecessary forward declaration

* Source/WebCore/platform/ios/wak/WAKAppKitStubs.mm: Renamed from Source/WebCore/platform/ios/wak/WAKAppKitStubs.m.
* Source/WebCore/platform/ios/wak/WAKClipView.mm: Renamed from Source/WebCore/platform/ios/wak/WAKClipView.m
* Source/WebCore/platform/ios/wak/WAKResponder.mm: Renamed from Source/WebCore/platform/ios/wak/WAKResponder.m.
* Source/WebCore/platform/ios/wak/WKUtilities.cpp: Renamed from Source/WebCore/platform/ios/wak/WKUtilities.c.
* Source/WebCore/platform/mac/WebCoreView.mm: Renamed from Source/WebCore/platform/mac/WebCoreView.m.

Make these files .cpp/.mm for consistency, to avoid module issues.

* Source/WebCore/platform/ios/wak/WebCoreThreadInternal.h:

Add a guard for a function that only is implemented on iOS.

* Source/WebCore/rendering/EventRegion.h:

Use NodeIdentifier include instead of Node.

Canonical link: <a href="https://commits.webkit.org/299308@main">https://commits.webkit.org/299308@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9864682e036b9c802d5392dd878d3c7ba6ed94ca

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118573 "25 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38254 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28905 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124750 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70632 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/fc669482-bfe5-4f94-b5c5-6deef11799b6) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120451 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38950 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46836 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89994 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59546 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/01c5e4be-3ba9-423e-89c5-a991a1016a44) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121526 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31023 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106308 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70498 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/f2e3a07d-72b9-4f2b-9e40-1c9aea7b2f29) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30078 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24419 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68407 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/110691 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100461 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24610 "Passed tests") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127811 "Failed to compile WebKit") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/117087 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45480 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34310 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/98628 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45844 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102529 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98412 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25018 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43852 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21846 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/41978 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45350 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/51028 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/145783 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44813 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37486 "Found 1 new JSC binary failure: testapi, Found 18596 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Array/toLocaleString.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/blue_1086262.js.default, ChakraCore.yaml/ChakraCore/test/ControlFlow/forInPrimitive.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default, ChakraCore.yaml/ChakraCore/test/Error/CallNonFunction.js.default, ChakraCore.yaml/ChakraCore/test/Error/stack2.js.default, ChakraCore.yaml/ChakraCore/test/Function/FuncBody.bug227901.js.default, ChakraCore.yaml/ChakraCore/test/Function/apply.js.default ... (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48160 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46500 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -47042,7 +47078,7 @@
 					};
 					93F198A508245E59001E9ABC = {
-						LastSwiftMigration = 1510;
+						LastSwiftMigration = 2630;
```
Not sure if this is a good idea?

rr-codes replied:
Ah, didn’t notice that. Will fix

rr-codes replied:
fixed!

pxlcoder commented on Configurations/CommonBase.xcconfig:
```diff
@@ -44,7 +44,7 @@ WK_GCC_OPTIMIZATION_LEVEL_SANITIZER_OVERRIDE_NO = $(WK_DEFAULT_GCC_OPTIMIZATION_
 
 WK_OS_UNFAIR_LOCK_INLINE_CFLAGS = $(WK_OS_UNFAIR_LOCK_INLINE_CFLAGS_$(USE_INTERNAL_SDK));
-WK_OS_UNFAIR_LOCK_INLINE_CFLAGS_YES = OS_UNFAIR_LOCK_INLINE=1
+WK_OS_UNFAIR_LOCK_INLINE_CFLAGS_YES = OS_UNFAIR_LOCK_INLINE=1 __ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0
```
How is `__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES` related to `os_unfair_lock`?

Seems like it should be a separate rule that's fed into `GCC_PREPROCESSOR_DEFINITIONS`?

rr-codes replied:
ah, ya I should just change the name of WK_OS_UNFAIR_LOCK_INLINE_CFLAGS , since the semantics and condition are the same for both

rr-codes committed:
[Swift in WebKit] Work towards modularizing WebCore private headers (Part 10)
https://bugs.webkit.org/show_bug.cgi?id=298068
rdar://159395829

Reviewed by Aditya Keerthi.

More changes to support modularization of WebCore; details inline.

* Configurations/CommonBase.xcconfig:

Add __ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0 to GCC_PREPROCESSOR_DEFINITIONS; this is needed so that it is 0
when building with modules (it's already defined as 0 in wtf/Platform.h, but that does not affect modules).

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:

Make some headers project instead of private.

* Source/JavaScriptCore/assembler/MacroAssemblerARMv7.h:
* Source/JavaScriptCore/heap/CellAttributes.h:
* Source/WebCore/contentextensions/ContentExtensionActions.h:
* Source/WebCore/crypto/CryptoAlgorithm.h:
* Source/WebCore/page/cocoa/WebTextIndicatorLayer.h:
* Source/WebCore/platform/graphics/Color.h:
* Source/WebCore/platform/graphics/ColorSpace.h:
* Source/WebCore/platform/graphics/ColorTypes.h:
* Source/WebCore/platform/graphics/DestinationColorSpace.h:
* Source/WebCore/platform/graphics/Path.h:
* Source/WebCore/platform/graphics/transforms/AffineTransform.h:
* Source/WebCore/style/values/text-decoration/StyleTextDecorationLine.h:
* Source/WebCore/svg/SVGTRefElement.cpp:
* Source/WebCore/svg/animation/SVGSMILElement.cpp:
* Source/WebCore/testing/Internals.cpp:
* Source/WebCore/workers/service/ServiceWorkerContainer.cpp:
* Source/WebCore/workers/service/ServiceWorkerContainer.h:

- Fix some quoted imports.
- Add missing includes

* Source/WTF/Scripts/modulemap-from-tapi-filelist.py:

All modules and sub-modules should always `export *` so that transitive dependency behavior remains intact.
Consequently, update the script to add this.

* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(WriteData):
* Source/WebCore/css/MediaQueryList.cpp:
* Source/WebCore/dom/EventListenerMap.cpp:
* Source/WebCore/dom/EventTarget.cpp:
* Source/WebCore/html/HTMLAttachmentElement.cpp:
* Source/WebCore/html/ImageDocument.cpp:
* Source/WebCore/html/PDFDocument.cpp:
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
* Source/WebCore/inspector/agents/InspectorIndexedDBAgent.cpp:
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.mm:
* Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm:
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginAnnotation.mm:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginPasswordField.mm:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginTextAnnotation.mm:
* Source/WebKitLegacy/mac/DOM/DOMNode.mm:

Use the AddEventListenerOptionsInlines.h file instead of AddEventListenerOptions.h

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

* Source/WebCore/accessibility/mac/CocoaAccessibilityConstants.h:

Fix typo and missed #define.

* Source/WebCore/dom/AddEventListenerOptions.h:
(WebCore::AddEventListenerOptions::AddEventListenerOptions): Deleted.
* Source/WebCore/dom/AddEventListenerOptionsInlines.h: Copied from Source/WebCore/dom/AddEventListenerOptions.h.
(WebCore::AddEventListenerOptions::AddEventListenerOptions):

Refactor the AddEventListenerOptions slightly to avoid an #include ordering issue due to a cycle.

* Source/WebCore/dom/EventTarget.h:

Include AddEventListenerOptions instead of using a forward declaration since the full type is needed for the variant
that uses it.

* Source/WebCore/html/canvas/WebGLExtension.h:
* Source/WebCore/html/canvas/WebGLExtensionAny.h:

Add missing newline and remove unnecessary forward declaration

* Source/WebCore/platform/ios/wak/WAKAppKitStubs.mm: Renamed from Source/WebCore/platform/ios/wak/WAKAppKitStubs.m.
* Source/WebCore/platform/ios/wak/WAKClipView.mm: Renamed from Source/WebCore/platform/ios/wak/WAKClipView.m
* Source/WebCore/platform/ios/wak/WAKResponder.mm: Renamed from Source/WebCore/platform/ios/wak/WAKResponder.m.
* Source/WebCore/platform/ios/wak/WKUtilities.cpp: Renamed from Source/WebCore/platform/ios/wak/WKUtilities.c.
* Source/WebCore/platform/mac/WebCoreView.mm: Renamed from Source/WebCore/platform/mac/WebCoreView.m.

Make these files .cpp/.mm for consistency, to avoid module issues.

* Source/WebCore/platform/ios/wak/WebCoreThreadInternal.h:

Add a guard for a function that only is implemented on iOS.

* Source/WebCore/rendering/EventRegion.h:

Use NodeIdentifier include instead of Node.

Canonical link: https://commits.webkit.org/299308@main
